### PR TITLE
Use concurrently_ from async instead of forkIO

### LINKED ISSUE #53: It gets stuck...
Today it's the ninth of january. The page was generated two days ago in the afternoon. For some reason it is stuck now. Lets find out why and fix it! 

- [ ] Add a timestamp when the data was last updated to the footer of the homepage.
- [ ] Log when the data is updated. 

denlasersig

### PR DESCRIPTION
**Pull request by adamse**
Maybe "fixes" #53.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -6,11 +6,11 @@ module Main
 where
 
 import           Control.Concurrent                       ( MVar
-                                                          , forkIO
                                                           , newMVar
                                                           , threadDelay
                                                           , tryPutMVar
                                                           )
+import qualified Control.Concurrent.Async                 as Async
```
Comment (Jassob) [inline app/Main.hs:13]: Why not skip the qualified import and simply import just `concurrently_` from Control.Concurrent.Async?

**Comment by Rembane** [comment]
@adamse I dig the code quality. But I must admit that my concurrent Haskell skills are a rusty. How does `concurrently_` solve the freezing problem?

**Comment by adamse** [comment]
So my guess about the freezing thing is that either the timer or the updater failed, since we were using forkIO these failures are not propagated to the main thread. Simon takes care of all the nasty details about getting this right in the async package.

So a failure in this new world would show up in the logs and let the service runner restart it. (So it's not a fix fix, but a fix in some sense :))

**Comment by Rembane** [comment]
Sweet! I like it when Simon takes care of all the weird bits.

File: app/Main.hs [hunk 2]
```diff
@@ -85,27 +85,27 @@ main =
...
-                  $ withFDHandler defaultBatchingOptions logHandle 1.0 80
-                  $ \logToHandle ->
-                      runReaderT (refreshAction upd) (ClientContext config mgr)
-                        `runLoggingT` ( logToHandle
-                                      . renderWithTimestamp
-                                          (formatTime
-                                            defaultTimeLocale
-                                            (iso8601DateFormat
-                                              (Just "%H:%M:%S")
+                Async.concurrently_
```
Comment (Jassob) [inline app/Main.hs:88]: Could we add something like `finally` or `bracket` to log when the server is shutdown?

**Review by Jassob — COMMENTED** [review]
Looks good!
I have some questions, though:

File: app/Main.hs (hunk 1 above)
Comment (adamse) [reply app/Main.hs:13]: I've taken to qualified imports since it's obvious at use site where the stuff comes from.

Comment (Jassob) [reply app/Main.hs:13]: I see your point, although I don't think you're missing that much when you list the imported stuff at the imports. 

My point was more that we should probably decide what we want to use so it's more or less consistent across the project :)

Comment (Jassob) [reply app/Main.hs:13]: But that is a discussion for somewhere else

File: app/Main.hs (hunk 2 above)
Comment (adamse) [reply app/Main.hs:88]: Exceptions will be printed to stdout iirc and the only way we can shut down is an exception.

**Review by Jassob — APPROVED** [review]
RTM!

**Pull request merged**